shared/libraries/native_streaming_protocol/src/native_streaming_server_handler.cpp

```diff
@@ -116,11 +116,12 @@ void NativeStreamingServerHandler::removeComponentSignals(const StringPtr& compo
 }
 
 bool NativeStreamingServerHandler::handleSignalSubscription(const SignalNumericIdType& signalNumericId,
-                                                            const std::string& signalStringId,
+                                                            const SignalPtr& signal,
                                                             bool subscribe,
                                                             const std::string& clientId)
 {
     std::scoped_lock lock(sync);
+    const auto signalStringId = signal.getGlobalId();
 
     if (subscribe)
     {
```

Comment:
I would place the signal permissions check here instead of in the session handler impl, like this:
```
...
   try
   {
      if (!sessionHandlers.at(clientId)->hasUserAccessToSignal(streamingManager.findRegisteredSignal(signalStringId)))
         throw NativeStreamingProtocolException("Access denied");
      ...
   }
```

Reply:
I was thinking about this approach, however it seems cleaner to me that all access protection is done in session handler. 

Otherwise I would argue that even the if from sendSignalAvailable() and sendSignalUnavailable() would need to be moved to the streaming server handler. Someone might to forget to wrap one function call leading to exposed signal. Anyone Implementing OnSignalSubscriptionCallback must also not forget to protect it with read access. It seems much cleaner to me that the callback is not even triggered if the user does not have access.